packages/flutter/test/widgets/routes_test.dart

```diff
@@ -892,3 +1019,68 @@ class DialogObserver extends NavigatorObserver {
     super.didPush(route, previousRoute);
   }
 }
+
+class _TestDialogRoute<T> extends PopupRoute<T> {
```

Comment:
You could potentially fold this and the next class into one by just calling "super.barrierCurve" if the barrierCurve provided in the constructor was null. It's up to you though.